fido/pronom/soap.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+FIDO: Format Identifier for Digital Objects
+
+Copyright 2010 The Open Preservation Foundation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+PRONOM format signatures SOAP calls.
+"""
+import sys
+import urllib.request
+import xml.etree.ElementTree as ET
+
+from fido import __version__
+ENCODING = 'utf-8'
+XML_PROC = '<?xml version="1.0" encoding="{}"?>'.format(ENCODING)
+TNA_DOMAIN = 'nationalarchives.gov.uk'
+PRONOM_HOST = 'www.{}'.format(TNA_DOMAIN)
+PRONOM_NS = 'http://pronom.{}'.format(TNA_DOMAIN)
+SIG_NS = 'http://{}/pronom/SignatureFile'.format(PRONOM_HOST)
+
+NS = {
+    'soap': 'http://schemas.xmlsoap.org/soap/envelope/',
+    'xsi': 'http://www.w3.org/2001/XMLSchema-instance',
+    'xsd': 'http://www.w3.org/2001/XMLSchema',
+    'pronom': PRONOM_NS,
+    'sig': SIG_NS
+}
+
+HEADERS = {
+    'Host': PRONOM_HOST,
+    'User-Agent': 'PRONOM UTILS v{0} (OPF)'.format(__version__),
+    'Content-type': 'text/xml; charset="UTF-8"'
+}
+
+def get_pronom_sig_version():
+    """
+    Get PRONOM signature version.
+
+    Return latest signature file version number as an int.
+    Raises an HTTPError if there are problems.
+    """
+    try:
+        tree = _get_soap_ele_tree('getSignatureFileVersionV1')
+        ver_ele = tree.find('.//pronom:Version/pronom:Version', NS)
+        return int(ver_ele.text)
+    except Exception as e:
+        sys.stderr.write('get_pronom_sig_version(): unknown error: {}'.format(str(e)))
+        raise e
```

Comment:
This is much nicer that what we used to have! Just a minor comment - I would probably do the writing to stderr from `fido.update_signatures` instead since that's where the CLI layer is. With the use case of the API in mind, most likely developers are not going to welcome any code that writes to their standard streams. Same idea applies to `get_pronom_signature`.
An alternative is to create a fido logger set up with the NullHandler so it's up to the user to install their own handlers (https://docs.python.org/2.7/howto/logging.html#library-config) - but that could be a task on its own.

Reply:
I think this is a good point, we should possibly remove output here and just throw exceps and log. These are quite "low level" functions and won't be used in just CLI calls going forward (am automating sig update so no release required). I'll look to address this.

Reply:
This is the one I've left for now, I think that the others are fixed and I went for Python 2 compatibility in the end @sevein

Reply:
So @sevein I've finished with this PR to my satisfaction now, even removed the use of `stderr` and sidestepped the Unicode encoding issue. Unit tests all pass on Python 2 and 3 as does real-life signature updating.